src/psyclone/psyir/nodes/literal.py

```diff
@@ -92,6 +96,20 @@ def __init__(self, value, datatype, parent=None):
         self._datatype = datatype
         self._value = value
 
+    @staticmethod
```

Comment:
Perhaps it would also make sense to create a LeafDataNode as well - depending on how many "<LeafNode>" classes there are? Again this assumes you don't go for the super-base-class-omatic solution.

Reply:
Look at the new implementation where the validate acts like a Leaf by default. Is this implementation ok?

Reply:
Yes.